### PR TITLE
add haul endpoint & functionality

### PR DESCRIPTION
**Pull request by mxtetrachord**

### CODE REVIEW

File: app/controllers/trash_blitz_controller.rb [hunk 1]
```diff
@@ -35,6 +35,20 @@ def get_pickups_csv
     end
   end
 
+  def submit_haul
```
Comment (PamelaM) [inline app/controllers/trash_blitz_controller.rb]: This find_and_create_by will only log one Pickup record if the haul contains multiples of the same item.  e.g. 10 Dunkin Donuts large coffee cups and lids

Comment (mxtetrachord) [reply app/controllers/trash_blitz_controller.rb]: fixed by returning the `Pickup`'s `id` to the frontend and using that to dedupe on the backend :)

File: app/controllers/trash_blitz_controller.rb [hunk 2]
```diff
@@ -15,6 +15,36 @@ def create_cleanup_event
...
+        id: itm["pickup_id"], brand: itm["brand"], latitude: itm["lat"], longitude: itm["long"],
+        material: Material.find(itm["variable_material"] ? itm["material"] : itm["material_id"]),
+        trash: Trash.find(itm["id"]),
+        session: Session.find(params[:session_id]),
+        cleanup_event: CleanupEvent.find_by(id: params[:cleanup_id])
+      })
+    end
+  end
+
+  def submit_haul
```
Comment (PamelaM) [inline app/controllers/trash_blitz_controller.rb]: submit_haul is duplicated...

Comment (mxtetrachord) [reply app/controllers/trash_blitz_controller.rb]: fixed!

**Pull request merged**